docs/doc/60-contributing/03-rfcs/20221013-iceberg-external-table.md

```diff
@@ -0,0 +1,257 @@
+---
+title: Iceberg External Table
+description: Accessing Apache Iceberg formatted data source as external table.
+---
+
+- RFC PR: [datafuselabs/databend#8215](https://github.com/datafuselabs/databend/pull/8215)
+- Tracking Issue: [datafuselabs/databend#8216](https://github.com/datafuselabs/databend/issues/8216)
+
+## Summary
+
+Iceberg is a widely supported table format among data lake-houses.
+This RFC describes how the iceberg external table will behave and how we will implement it.
+
+## Motivation
+
+The [Apache Iceberg](https://iceberg.apache.org) is a table format widely used in data lake-houses, it offers a more complete logical view of databases and has much higher performance of accessing as external tables, since users don't need to know about partitions and less indexing through files when accessing is required comparing to `Hive`.
+
+The splitted and well-defined structure of `Iceberg` also makes concurrent accesssing and version managementing on data sources safer, saver and convienient.
+
+![stack of Apache Iceberg](https://iceberg.apache.org/img/iceberg-metadata.png)
+
+Supporting Iceberg will empower databend as a open data lake, giving it more OLAP abilities.
+
+## Guide-level explanation
+
+### Create Iceberg Table
+
+To use Iceberg external table, users need to have an iceberg storage with neccessary permission established. Then they can establish a external table with `ICEBERG` engine:
+
+```sql
+CREATE EXTERNAL TABLE [IF NOT EXISTS] [db.]table_name
```

Comment:
We can remove the `EXTERNAL`, and treat the iceberg as a storage engine.

Reply:
Databend will support many storage engines(Now support FUSE/HIVE+HDFS), it would be better to treat Iceberg as a new storage engine, not an external table.